lib/components/metrics.js

```diff
@@ -0,0 +1,293 @@
+        namespace: "process",
+        name: "cpu_user_seconds_total",
+        help: "Total user CPU time spent in seconds",
+    });
+    var cpu_system_gauge = this.addGauge({
+        namespace: "process",
+        name: "cpu_system_seconds_total",
+        help: "Total system CPU time spent in seconds",
+    });
+
+    this.addCollector(function() {
+        var usage = process.memoryUsage();
+
+        rss_gauge.set({}, usage.rss);
+        heap_size_gauge.set({}, usage.heapTotal);
+        heap_used_gauge.set({}, usage.heapUsed);
+
+        var stats = _read_proc_self_stat();
+
+        // CPU times in ticks
+        var utime_secs = stats[11] / TICKS_PER_SEC;
+        var stime_secs = stats[12] / TICKS_PER_SEC;
+
+        cpu_gauge.set({}, utime_secs + stime_secs);
+        cpu_user_gauge.set({}, utime_secs);
+        cpu_system_gauge.set({}, stime_secs);
+
+        // Virtual memory size
+        vsz_gauge.set({}, stats[20]);
+    });
+
+    this.addGauge({
+        namespace: "process",
+        name: "open_fds",
+        help: "Number of open file descriptors",
+        refresh: function(gauge) {
+            var fds = fs.readdirSync("/proc/self/fd");
+
+            // subtract 1 due to readdir handle itself
+            gauge.set(null, fds.length - 1);
+        }
+    });
+
+    this.addGauge({
+        namespace: "process",
+        name: "max_fds",
+        help: "Maximum number of open file descriptors allowed",
+        refresh: function(gauge) {
+            var limits = fs.readFileSync("/proc/self/limits");
+            limits.toString().split(/\n/).forEach((line) => {
+                if (!line.match(/^Max open files /)) return;
+
+                // "Max", "open", "files", $SOFT, $HARD, "files"
+                gauge.set({}, line.split(/\s+/)[3]);
+            });
+        }
+    });
+
+    // This value will be constant for the lifetime of the process
+    this.addGauge({
+        namespace: "process",
+        name: "start_time_seconds",
+        help: "Start time of the process since unix epoch in seconds",
+    }).set({}, _calculate_process_start_time());
+
+    this.refresh();
+};
+
+function _read_proc_self_stat() {
+    var stat_line = fs.readFileSync("/proc/self/stat")
+        .toString().split(/\n/)[0];
+    // Line contains PID (exec_name) bunch of stats here...
+    return stat_line.match(/\) +(.*)$/)[1].split(" ");
+}
+
+function _calculate_process_start_time() {
+    // The 'starttime' field in /proc/self/stat gives the number of CPU ticks
+    //   since machine boot time that this process started.
+    var stats = _read_proc_self_stat();
+    var starttime_sec = stats[19] / TICKS_PER_SEC;
+
+    var btime_line = fs.readFileSync("/proc/stat")
+        .toString().split(/\n/).filter((l) => l.match(/^btime /))[0];
+    var btime = Number(btime_line.split(" ")[1]);
+
+    return btime + starttime_sec;
+}
+
+Metrics.prototype.refresh = function() {
+    this._collectors.forEach((f) => f());
```

Comment:
I think rather than `try {} catch {}` wrapping the collector functions, better would be to only even enable those metrics if `/proc/self/stat` is found to be working at construction time.

Done in 84b5cb7